app/controllers/social_auth_controller.rb

```diff
@@ -1,81 +1,60 @@
-class SocialAuthController < BaseController
+class SocialAuthController < ApplicationController
   include LoginSignupHelper
 
   fine_print_skip :general_terms_of_use, :privacy_policy
 
   skip_before_action :authenticate_user!, only: [ :oauth_callback, :confirm_oauth_info ]
 
-  before_action :restart_signup_if_missing_unverified_user, only: [
-    :confirm_oauth_info, :confirm_social_info_form
-  ]
-
   # Log in (or sign up and then log in) a user using a social (OAuth) provider
   def oauth_callback
+    @user = current_user
     if signed_in? && user_signin_is_too_old?
       reauthenticate_user!
     else
       handle_with(
         OauthCallback,
-        logged_in_user: signed_in? && current_user,
         success: lambda {
           authentication = @handler_result.outputs.authentication
           user = @handler_result.outputs.user
 
-          if user.student? && !user.activated?
-            # Not activated means signup.
-            # Only students can sign up with a social network.
-            unverified_user = ensure_unverified_user(user)
+          sign_in!(user)
 
-            save_unverified_user(unverified_user.id)
+          unless user.activated? # needs social confirmation page for info
             @first_name = user.first_name
-            @last_name = user.last_name
-            @email = @handler_result.outputs.email
+            @last_name  = user.last_name
+            @email      = @handler_result.outputs.email
             security_log(:student_social_sign_up, user: user, authentication_id: authentication.id)
-            # must confirm their social info on signup
-            render :confirm_social_info_form and return # TODO: if possible, update the route/path to reflect that this page is being rendered
+            redirect_to confirm_oauth_info_path and return
```

Comment:
This may have to be a before_action too? Otherwise they can escape since the session is already set